reactor-core/src/main/java/reactor/core/publisher/OptimisticEmitFailureHandler.java

```diff
@@ -0,0 +1,44 @@
+package reactor.core.publisher;
+
+import java.time.Duration;
+
+/**
+ * @author Animesh Chaturvedi
+ * This implementation deals with the transient failure in case of emit from
+ * sink because of competing threads. Busy looping is used as an approach instead
+ * of {@link Sinks.EmitFailureHandler#FAIL_FAST} with specified duration
+ */
+public class OptimisticEmitFailureHandler implements Sinks.EmitFailureHandler {
```

Comment:
Can you please make this implementation package-private? I would even prefer it is moved under `Sinks` so that all Sink-related interfaces and implementations are under the same umbrella.